pxemgr/etcd_discovery_handlers.go

```diff
@@ -104,14 +106,29 @@ func (mgr *pxeManagerT) etcdDiscoveryProxyRequest(r *http.Request) (*http.Respon
 	if err != nil {
 		return nil, err
 	}
+	var transport = http.DefaultTransport
 
-	for i := 0; i <= 10; i++ {
-		u := url.URL{
-			Scheme:   "http",
-			Host:     mgr.etcdEndpoint,
-			Path:     path.Join("v2", "keys", "_etcd", "registry", strings.TrimPrefix(r.URL.Path, "/etcd")),
-			RawQuery: r.URL.RawQuery,
+	if strings.HasPrefix(mgr.etcdEndpoint, "https") && mgr.etcdCAFile != "" {
```

Comment:
When you properly parse the URL you can directly use `u.Scheme` to check if it is the scheme you want to check for.